transport/src/main/java/io/netty/channel/oio/AbstractOioByteChannel.java

```diff
@@ -40,46 +40,23 @@
             " (expected: " + StringUtil.simpleClassName(ByteBuf.class) + ", " +
             StringUtil.simpleClassName(FileRegion.class) + ')';
 
-    private volatile boolean inputShutdown;
-
     /**
      * @see AbstractOioByteChannel#AbstractOioByteChannel(Channel)
      */
     protected AbstractOioByteChannel(Channel parent) {
         super(parent);
     }
 
-    protected boolean isInputShutdown() {
-        return inputShutdown;
-    }
-
     @Override
     public ChannelMetadata metadata() {
         return METADATA;
     }
 
-    /**
-     * Check if the input was shutdown and if so return {@code true}. The default implementation sleeps also for
-     * {@link #SO_TIMEOUT} milliseconds to simulate some blocking.
-     */
-    protected boolean checkInputShutdown() {
-        if (inputShutdown) {
```

Comment:
@normanmaurer - making `isInputShutdown()` abstract and removing this method is an API change ... i can be more strict with the API but this will change will only be for 4.1, and its OIO... let me know if you think I should spend more effort here to preserve the API